src/org/deepsymmetry/bytefield/core.cljs

```diff
@@ -531,57 +531,85 @@
-  the preceding box)."
+  the preceding box).
+  
+  The background can be filled with a color passed with `:fill`."
   ([]
    (draw-gap nil nil))
   ([label]
    (draw-gap label nil))
   ([label attr-spec]
-   (let [{:keys [height gap edge gap-style box-above-style min-label-columns]
+   (let [attrs (eval-attribute-spec attr-spec)]
+   (let [{:keys [height gap edge gap-style box-above-style min-label-columns fill]
           :or   {height            70
                  gap               10
                  edge              15
                  gap-style         (eval-attribute-spec :dotted)
                  box-above-style   (eval-attribute-spec :box-above)
-                 min-label-columns 8}} (eval-attribute-spec attr-spec)
+                 min-label-columns 8}} attrs
 
          column (:column @@('diagram-state @*globals*))
          boxes  @('boxes-per-row @*globals*)]
      (if label
        ;; We are supposed to draw a label.
        (if (<= min-label-columns (- boxes column))
-         (draw-box label [{:span (- boxes column)} box-above-style]) ; And there is room for it on the current line.
+         (draw-box label [{:span (- boxes column)} box-above-style (if (nil? fill) {} {:fill fill})]) ; And there is room for it on the current line.
          (do ; The label doesn't fit on the current line.
-           (draw-box nil [{:span (- boxes column)} box-above-style]) ; Finish off current line with emptiness.
+           (draw-box nil [{:span (- boxes column)} box-above-style (if (nil? fill) {} {:fill fill})]) ; Finish off current line with emptiness.
            (auto-advance-row nil)
-           (draw-box label [{:span boxes :borders #{:left :right}}]))) ; Put the label on its own line.
+           (draw-box label [{:span boxes :borders #{:left :right}} (if (nil? fill) {} {:fill fill})]))) ; Put the label on its own line.
        ;; We are not supposed to draw a label, so just finish the current line if needed.
        (when (not= column boxes)
-         (draw-box nil [{:span (- boxes column)} :box-above]))) ; Finish off current line with emptiness.
+         (draw-box nil [{:span (- boxes column)} :box-above (if (nil? fill) {} {:fill fill})]))) ; Finish off current line with emptiness.
 
      ;; Move on to a new row to draw the gap.
      (auto-advance-row nil)
      (let [y      (diagram-height)
            top    (+ y edge)
            left   @('left-margin @*globals*)
            right  (+ left (diagram-width))
-           bottom (+ y (- height edge))]
-       (draw-line left y left top)
-       (draw-line right y right top)
-       (draw-line left top right (- bottom gap) gap-style)
-       (draw-line right y right (- bottom gap))
-       (draw-line left (+ top gap) right bottom gap-style)
-       (draw-line left (+ top gap) left bottom)
-       (draw-line left bottom left (+ y height))
-       (draw-line right bottom right (+ y height)))
+           bottom (+ y (- height edge))
+           upper-trapezoid { 
```

Comment:
And in my own pass through, I corrected the non-Clojure brace styles in these places, it’s not idiomatic to have an open or close brace by itself, so I pulled the `:top-left` definition up to the preceding line, and pulled the close brace on 577 up to touch the other closing paren/bracket on line 576 (in fact Emacs does this for me automatically when I type a close paren at the end of line 576 in structural editing mode). Similarly with the other opening and closing delimiters, like line 598 all gets pulled up to touch `fill`, the close parne on 597 gets pulled up to 596, but `:fill fill ))])` is its own line still.